### DISCUSSION TITLE
Custom Constraint Validation

### DISCUSSION THREAD

**Discussion by KarunBakshi** [networknt/json-schema-validator#1118] [Q&A]
Hello,
  New to this project so perhaps a basic question: Can I do custom constraint validation with json-schema-validator? For example, is there a plug-in mechanism to specify tags and code that validates against them. I know there is a parse tree walker, but not sure whether it can be used for this purpose, and if so, how? Thanks!

Karun

**Comment by justin-tay** [comment]
You can create custom keywords, formats and vocabularies. 

See https://github.com/networknt/json-schema-validator/blob/master/doc/custom-meta-schema.md.
